hub/api/dataset.py

```diff
@@ -190,6 +183,16 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def _load_meta(self):
```

Comment:
I'm with dyllan on this one. Whenever we do `self.meta`, we'll expect it to be up-to-date. This deferred style is highly prone to error; the `@property` serves as a getter to always fetch from storage. 

If we're concerned for performance, we should integrate it at the `Meta` class level, so that we can guarantee that any modification will trigger a reload.

Reply:
I would follow a similar routine for `self.tensors`, as otherwise it might get hairy.

Reply:
Should this be done right now?
For every case except for transforms, the meta should be loaded from the version stored in memory. It's only the case of transforms where I modify underlying meta files without interacting with the dataset api that it should be fetched again from memory. I can create a task in the backlog to address this properly later.

Reply:
@AbhinavTuli okay. I'm not sure I completely understand how we're handling the meta, but as long as the following are true:
- The user is always using the up-to-date meta.
- Any user modification to the meta gets reflected in the storage
- Any underyling modification bubbles up to the user's copy of the meta

Then I'm happy.

Reply:
bubbling doesn't happen yet, should add to backlog

Reply:
added to backlog.